returnn/tf/layers/basic.py

```diff
@@ -859,39 +859,48 @@ class SliceNdLayer(_ConcatInputLayer):
   def __init__(self, start, size, min_size=None, **kwargs):
     """
     :param LayerBase start: (B,...)
-    :param int|None size: if None, it uses the max possible size, and it becomes a dynamic axis
+    :param int|LayerBase|None size: if None, it uses the max possible size,
+      and it becomes a dynamic axis.
     :param int|None min_size: if size is None, but we want to have a min-size
     """
     super(SliceNdLayer, self).__init__(**kwargs)
     from returnn.tf.util.basic import where_bc, expand_multiple_dims
-    x = self.input_data.copy_as_batch_major()
-    seq_lens = x.get_sequence_lengths() if x.is_time_axis_dynamic() else None  # (B,) or None
+    from returnn.tf.util.data import Data
+    x = self.input_data.copy()
+    seq_lens_data = x.get_time_dim_tag().dyn_size_ext  # (B,) or None
     self.start = start
-    start_data = start.output.copy_as_batch_major()  # e.g. (B,) or (B,T)
+    self.size = size
+    start_data = start.output.copy()  # e.g. (B,) or (B,T)
```

Comment:
We potentially have 3 sources here for the shape of the positions:
`seq_lens_data`, `start` and `size`.
I think we should use `get_common_data` on all three, and then store sth like `common_data = Data.get_common_data([seq_lens_data, start, size.output if isinstance(size, LayerBase) else None])`, and below, use `copy_compatible_to(common_data, ...)`.

Reply:
So `get_common_data` does not allow the list items to be `None`. Can I instead do:

```
data_objs = [start_data]
data_objs += [size.output] if isinstance(size, LayerBase) else []
data_objs += [seq_lens_data] if isinstance(seq_lens_data, Data) else []
common_data = Data.get_common_data(data_objs)?
```

Reply:
Ah yes.